extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -651,8 +651,9 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
     });
 
     let imageIdsToSet = imageIds;
-    const overlayProcessingResult = this._processExtraDisplaySetsForViewport(viewport);
-    imageIdsToSet = overlayProcessingResult?.imageIds ?? imageIdsToSet;
+    const overlayProcessingResults = this._processExtraDisplaySetsForViewport(viewport);
+    // Since we have a stack viewport, all of the imageIds should be the same for each overlay, no?
```

Comment:
@sedghi Is this assumption about the imageIds correct for a stack viewport? If not, what should I do instead?

Reply:
each segmentation (labelmap) in stack settings will have its own sets of imagesIds. But for instance if we are on slice 150 in CT , only 1 or more (overlapping) of those imageIds can get rendererd, but that is done automatically by cornerstone. So here you should just add the segmentation representation with those imageIds and add them

Reply:
@sedghi  So are you suggesting that we don't even need this line? I don't think I completely understand.